### ISSUE TITLE
Update server names stored in Raft state

### ISSUE THREAD

**Issue by timmaxw** [rethinkdb/rethinkdb#4252]
After #3897 we store a non-authoritative copy of server names in the Raft state alongside the server IDs, so we can still know the server names even if the server disconnects. The Raft leader should automatically update this copy whenever a server is renamed.

**Comment by VeXocide** [comment]
In CR 2992 by @timmaxw.

**Comment by VeXocide** [comment]
Merged into `next` via commit 87909f93dcf4bd072ecefd495c68b6b3209d1eb0.
